foodfetch: init at 0.1.1

noahfraiture opened the pull request:
Foodfetch is a simple fetch tool to get recipes with a picture of the recipe in ascii.
https://github.com/noahfraiture/foodfetch.git

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

https://github.com/NixOS/nixpkgs/pull/396167 : minor update changing a panic in a log https://github.com/wavetermdev/waveterm/compare/v0.11.1...v0.11.2

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

noahfraiture committed:
foodfetch: init at 0.1.1

Apply suggestions from code review

Co-authored-by: emaryn <197520219+emaryn@users.noreply.github.com>

fix

Update pkgs/by-name/fo/foodfetch/package.nix

Co-authored-by: awwpotato <153149335+awwpotato@users.noreply.github.com>

Update pkgs/by-name/fo/foodfetch/package.nix

Co-authored-by: awwpotato <153149335+awwpotato@users.noreply.github.com>

noahfraiture commented:
Thank you for your help in this review !

haylinmoore commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 396172`

Logs: https://github.com/haylinmoore/nixpkgs-review-gha/actions/runs/14286201324


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>


---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>

awwpotato commented on pkgs/by-name/fo/foodfetch/package.nix:
```diff
@@ -0,0 +1,32 @@
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+
```
```suggestion

  doInstallCheck = true;
  nativeInstallCheck = [ versionCheckHook ];
  
  passthru.updateScript = nix-update-script { };

```

0xda157 reviewed (commented):
otherwise lgtm

awwpotato commented on pkgs/by-name/fo/foodfetch/package.nix:
```diff
@@ -0,0 +1,39 @@
+    changelog = "https://github.com/noahfraiture/foodfetch/releases/tag/v${finalAttrs.version}";
+    description = "Yet another fetch to quickly get recipes";
+    homepage = "https://github.com/noahfraiture/foodfetch";
+    license = lib.licenses.mit;
```
you also need to add yourself as a maintainer, both here and in [the maintainers list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#how-to-become-a-maintainer)

noahfraiture replied:
oh sorry, I found PR without that, I guess they were already maintainers and I missed it

awwpotato commented:
You should make adding yourself as a maintainer a separate commit titled `maintainers: add noahfraiture`. Aside from that lgtm

a-kenji commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 396172`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>

a-kenji reviewed (approved):
Agree that the commits should be split up as per the contribution guidelines.
Other than that :+1: .

noahfraiture committed:
maintainers: add noahfraiture

noahfraiture commented:
Hello, is there still something missing ? I can see that the approval has been removed by @wegank

awwpotato commented:
> I can see that the approval has been removed by @wegank

it was automatically removed by https://github.com/nix-community/label-approved because there were changes since the last commit.

wegank commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 396172`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foodfetch</li>
  </ul>
</details>

noahfraiture commented:
> > I can see that the approval has been removed by @wegank
> 
> it was automatically removed by https://github.com/nix-community/label-approved because there were changes since the last approval.

Oh ok, thank you for the explanation
